### PR TITLE
ASGI: Update middleware docs

### PR DESCRIPTION
**Pull request by kgriffs**
# Summary of Changes

Update the middleware docs to better address the new ASGI implementation.

# Related Issues

#1358

### CODE REVIEW

File: docs/api/middleware.rst [hunk 1]
```diff
@@ -66,6 +66,124 @@ Falcon's middleware interface is defined as follows:
...
+
+    class ExampleComponent:
+        def process_request(self, req, resp):
+            """Process WSGI request using synchronous logic.
+
+            Note that req and resp are instances of falcon.Request and
+            falcon.Response, respectively
+            """
+
+        async def def process_request_async(self, req, resp):
```
Comment (vytas7) [inline docs/api/middleware.rst]: Syntax: `async def def`

File: docs/api/middleware.rst [hunk 2]
```diff
@@ -66,6 +66,124 @@ Falcon's middleware interface is defined as follows:
...
+            """Process ASGI request using asynchronous logic.
+
+            Note that req and resp are instances of falcon.asgi.Request and
+            falcon.asgi.Response, respectively
```
Comment (vytas7) [inline docs/api/middleware.rst]: A full stop (period) missing.

File: docs/api/middleware.rst [hunk 3]
```diff
@@ -66,6 +66,124 @@ Falcon's middleware interface is defined as follows:
...
+the following example:
+
+.. code:: python
+
+    class ExampleComponent:
+        def process_request(self, req, resp):
+            """Process WSGI request using synchronous logic.
+
+            Note that req and resp are instances of falcon.Request and
+            falcon.Response, respectively
```
Comment (vytas7) [inline docs/api/middleware.rst]: A full stop (period) missing.

File: docs/api/middleware.rst (hunk 1 above)
Comment (kgriffs) [reply docs/api/middleware.rst]: 🤦‍♂

File: docs/api/middleware.rst (hunk 2 above)
Comment (kgriffs) [reply docs/api/middleware.rst]: fixed

File: docs/api/middleware.rst (hunk 3 above)
Comment (kgriffs) [reply docs/api/middleware.rst]: fixed

File: docs/api/middleware.rst (hunk 1 above)
Comment (kgriffs) [reply docs/api/middleware.rst]: fixed

**Pull request merged**